### PR TITLE
1120: Hide Created and Sold when collector has 0 created NFTs

### LINKED ISSUE #1120: Hide Created and Sold when collector has 0 created NFTs
Hide Created and Sold when query resolves that Artist profile did not issued any NFT

image

Test case https://kodadot.xyz/rmrk/u/DgCNKpvqeg29wsMsUDLcYP7Kqcc4pW963F7mj2uRe8Kk6Jm?tab=nft

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the KodaDot NFT gallery.

👇 Do a quick check before the merge. 

### PR type
- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

### Before submitting Pull Request, please make sure:
- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted screenshot of demonstrated change in this PR

### Optional
- [ ] I've tested PR on mobile and everything works
- [ ] I found edge cases

### What's new?
- [x] PR closes #1120
- [ ] <brief_description_of_what_I've_added>

### Had issue bounty label ?
- [x] Fill up your KSM address: 
[Payout](https://kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

### Community participation
- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot
- [ ] My fix has changed **something** on UI, a screenshot for others, is best to understand changes.

### CODE REVIEW

File: src/views/Profile.vue [hunk 1]
```diff
@@ -52,7 +52,7 @@
         expanded
         size="is-medium"
       >
-        <b-tab-item value="nft">
+        <b-tab-item value="nft" :headerClass="!totalCreated ? 'is-hidden' : '' ">
```
Comment (roiLeo) [inline src/views/Profile.vue:55]:
```suggestion
        <b-tab-item value="nft" :headerClass="{'is-hidden': !totalCreated}" >
```
small refacto

File: src/views/Profile.vue [hunk 2]
```diff
@@ -82,7 +82,7 @@
             v-model="currentCollectionPage"
           />
         </b-tab-item>
-        <b-tab-item value="sold">
+        <b-tab-item value="sold" :headerClass="!totalSold ? 'is-hidden' : '' ">
```
Comment (roiLeo) [inline src/views/Profile.vue:85]: same here

**Review by roiLeo — CHANGES_REQUESTED** [review]

**Review by roiLeo — CHANGES_REQUESTED** [review]
Tested on /rmrk/u/F5DaGLPBkroG9tYGZPaaAaQWqzkNgJgjrz17FAmk1EVNKn9
It's not displaying "Sold" tab while this user has already sold nfts

**Comment by yangwao** [comment]
Yeah, it's good to show up test suite where you testing it, probably you can add it to `TESTSUITE.md` file for case why

**Comment by prachi00** [comment]
> Tested on /rmrk/u/F5DaGLPBkroG9tYGZPaaAaQWqzkNgJgjrz17FAmk1EVNKn9 It's not displaying "Sold" tab while this user has already sold nfts

@roiLeo so the issue now is that how do we know the number of sold items? the graphql query only hits once the tab is actually clicked. 
Possible solution:
hit the graphql query for **nftListSold**  once the page loads to get the number of sold items

**Comment by roiLeo** [comment]
> > Tested on /rmrk/u/F5DaGLPBkroG9tYGZPaaAaQWqzkNgJgjrz17FAmk1EVNKn9 It's not displaying "Sold" tab while this user has already sold nfts
> 
> @roiLeo so the issue now is that how do we know the number of sold items? the graphql query only hits once the tab is actually clicked. Possible solution: hit the graphql query for **nftListSold** once the page loads to get the number of sold items

It will be the same for Collected & Created.
Possible solution could be to create a new query only for the stats (Created/Collected/Collection/Sold)

**Comment by prachi00** [comment]
> > > Tested on /rmrk/u/F5DaGLPBkroG9tYGZPaaAaQWqzkNgJgjrz17FAmk1EVNKn9 It's not displaying "Sold" tab while this user has already sold nfts
> > 
> > 
> > @roiLeo so the issue now is that how do we know the number of sold items? the graphql query only hits once the tab is actually clicked. Possible solution: hit the graphql query for **nftListSold** once the page loads to get the number of sold items
> 
> It will be the same for Collected & Created. Possible solution could be to create a new query only for the stats (Created/Collected/Collection/Sold)

@roiLeo okay, can you tell me more about how to do this? like is there already an api that is giving out these details?

**Comment by prachi00** [comment]
> > @roiLeo okay, can you tell me more about how to do this? like is there already an api that is giving out these details?
> 
> Well, I'm not an expert on this field but you could try something with graphql to get multiple totalCount. I would do my tests on [graphqlbin](https://www.graphqlbin.com/v2/new) with magic SUBQUERY_URL

@roiLeo so I tried doing it using the spotlight api, I still couldn't find two issues tho:
1. Finding the count of **collected**
2. Using IDB, since I couldn't find any data for spotlight already there

File: src/views/Profile.vue [hunk 3]
```diff
@@ -287,6 +290,29 @@ export default class Profile extends Vue {
...
+        variables: {
+          account: this.id,
+        },
+      })
+
+      const {
+        data: { collectionEntities },
+      } = stats
+
+      const [data] :any = spotlightAggQuery(
```
Comment (roiLeo) [inline src/views/Profile.vue:304]:
```suggestion
      const [data]: any = spotlightAggQuery(
```

File: src/views/Profile.vue [hunk 4]
```diff
@@ -332,6 +358,12 @@ export default class Profile extends Vue {
     }
   }
 
+  protected async handleSoldResult({ data }: any) {
+    if (data) {
+      this.totalSold = data.nFTEntities.totalCount
+    }
+  }
+
```
Comment (roiLeo) [inline src/views/Profile.vue:366]: this could be removed as it's not used anymore

**Review by roiLeo — APPROVED** [review]
Tested on: 
- /rmrk/u/CwXa36Dddsm4h7T1LDnXvtdTFmSfUg7ZVygZ7mv8m6DE19s
- /rmrk/u/FUEGFWoPPtX4XkRzEnZnMgwGKENscHXAiEZy6kcgaDz1BY6
- /rmrk/u/EfzipPaWQRa8p7RtfKqw2QUFrZDSbHcXVjuD9XEDJKaBw9a

We might need to check if there is a significant increase in loading time by adding this query

**Comment by vikiival** [comment]
To be honest I do no like the fix

1. It's a super big query
2. sold => currentOwner != issuer (yeah not great)

**Review by vikiival — CHANGES_REQUESTED** [review]
Super big query

**Comment by prachi00** [comment]
> To be honest I do no like the fix
> 
> 1. It's a super big query
> 2. sold => currentOwner != issuer (yeah not great)
> 
> My approach to this: just check if the first query (created) returns empty array and emit the change to the parent component :)
> 
> cc @roiLeo @prachi00

@vikiival you mean the query for sold items?

**Comment by yangwao** [comment]
> just check if the first query (created) returns empty array and emit the change to the parent component :)

yes, i've proposed this as well :)

**Comment by vikiival** [comment]
https://github.com/kodadot/nft-gallery/compare/main...1120-hide-created-when-empty

**Comment by yangwao** [comment]
> cc @roiLeo @prachi00
> 
> [main...1120-hide-created-when-empty](https://github.com/kodadot/nft-gallery/compare/main...1120-hide-created-when-empty)

@prachi00 let's pick up this, it easiest way to resolve it and does the job, happy to proceed with payout :)

**Comment by prachi00** [comment]
> > cc @roiLeo @prachi00
> > [main...1120-hide-created-when-empty](https://github.com/kodadot/nft-gallery/compare/main...1120-hide-created-when-empty)
> 
> @prachi00 let's pick up this @vikiival solution, it easiest way to resolve it and does the job, happy to proceed with payout :)

Did the changes, not sure if its right or not tho. Also PS had to force push (with lease) since I had to delete my previous commit

**Review by roiLeo — APPROVED** [review]

**Comment by yangwao** [comment]
Looks good BUT when I open https://deploy-preview-1133--nftkodadot.netlify.app/rmrk/u/DgCNKpvqeg29wsMsUDLcYP7Kqcc4pW963F7mj2uRe8Kk6Jm

It just keeps hanging on collected. 
image


I suggest you should watch totalcollections and if zero then set active tab to collected

**Comment by prachi00** [comment]
> Looks good BUT when I open https://deploy-preview-1133--nftkodadot.netlify.app/rmrk/u/DgCNKpvqeg29wsMsUDLcYP7Kqcc4pW963F7mj2uRe8Kk6Jm
> 
> It just keeps hanging on collected. image
> 
> I suggest you should watch totalcollections and if zero then set active tab to collected

@yangwao done

**Review by vikiival — APPROVED** [review]

**Comment by yangwao** [comment]
nice works for me now!
sent payout https://kusama.subscan.io/extrinsic/10177383-7

**Pull request merged**